### PR TITLE
[2.7][Form] placeholder option: replace "in favor" misuses

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7+ |
| Fixed tickets | - |

### CODE REVIEW

**Comment by xabbuh** [comment]
:+1:

**Comment by WouterJ** [comment]
For a non-native speaker as me, what is the reason behind this PR? It looks like "in favor of" is very commonly used in the docs and code repository.

**Comment by ogizanagi** [comment]
@WouterJ : I'm a non-native english speaker too, and I find this very confusing:
> The ``placeholder`` option was introduced in Symfony 2.6 in favor of ``empty_value``

The left operand of `in favor of` (if I may say so ^^') usually is the value we'll replace. 
The right operand is the new value. 

But in this sentence, it is misused: the `placeholder` is the new option. We'll don't abandon it in favor of the `empty_value` option.

So, to me, `in replacement` makes more sense.
By the way, AFAIK this is the only place where `in favor` is misused. Other occurences are perfectly valid.
For instance:
> deprecated the `intention` firewall listener setting in favor of the `csrf_token_id`

The `intention` option is the one deprecated. Instead of using, we should use `csrf_token_id`

**Comment by xabbuh** [comment]
@weaverryan What do you think?

File: cookbook/form/create_custom_field_type.rst [hunk 1]
```diff
@@ -257,7 +257,7 @@ the gender codes were stored in configuration or in a database? The next
 section explains how more complex field types solve this problem.
 
 .. versionadded:: 2.6
-    The ``placeholder`` option was introduced in Symfony 2.6 in favor of
+    The ``placeholder`` option was introduced in Symfony 2.6 in replacement of
```
Comment (weaverryan) [inline cookbook/form/create_custom_field_type.rst]: Yea, I definitely see your point - "in favor of" usually means that the item to follow is the new, good one.

But "in replacement of" is kind of an awkward phrase. What about:

```rst
The ``placeholder`` option was introduced in Symfony 2.6 and replaces
``empty_value``, which is available prior to 2.6.
```

Comment (ogizanagi) [inline cookbook/form/create_custom_field_type.rst]: Thanks @weaverryan ! I updated the PR accordingly.

**Comment by xabbuh** [comment]
Thank you @ogizanagi.

**Pull request merged**